File: mobile/android/sdk/src/main/kotlin/com/rtcstack/sdk/RTCstack.kt
package com.rtcstack.sdk

import android.content.Context

/**
 * Entry point for the RTCstack Android SDK. Mirrors `createCall` from `@rtcstack/sdk`.
 *
 * ```kotlin
 * val call = RTCstack.createCall(context, CallOptions(token = jwt, url = wssUrl))
 * call.connect()
 * ```
 *
 * The SDK only ever holds a LiveKit JWT + WSS URL — never the RTCstack API key/secret.
 * Mint tokens on your backend (POST /v1/token) and pass them in.
 */
public object RTCstack {
    public fun createCall(context: Context, options: CallOptions): Call =
        Call(context.applicationContext, options)
}

File: mobile/ios/Sources/RTCstackKit/RTCstack.swift
import Foundation

/// Entry point for the RTCstack iOS SDK. Mirrors `createCall` from `@rtcstack/sdk`.
///
/// ```swift
/// let call = RTCstack.createCall(.init(token: jwt, url: wssURL))
/// try await call.connect()
/// ```
///
/// The SDK only ever holds a LiveKit JWT + WSS URL — never the RTCstack API key/secret.
/// Mint tokens on your backend (POST /v1/token) and pass them in.
public enum RTCstack {
    @MainActor
    public static func createCall(_ options: CallOptions) -> Call {
        Call(options: options)
    }
}
